### PR TITLE
Fixed issue #1415 

### PR DESCRIPTION
**Pull request by msanguineti**
and also avoid checking for case in Windows file/folders names

Signed-off-by: msanguineti <mirco@sanguineti.gr>

### CODE REVIEW

**Comment by msanguineti** [comment]
Hello @cescoffier,

I see that you fixed #1415 but may I suggest to have a look at my code since I take care of allowing OS/X and *nix to pass Windows style path (it might be important in mixed environments where developers write scripts in either systems) but more importantly I do not check for case sensitivity in Windows files/folders when the Watcher tries to match a change. Also I wrote a couple of tests for those cases.

Regards,
Mirco

**Comment by cescoffier** [comment]
Can you rebase your commits on mine to see the difference ?

**Comment by msanguineti** [comment]
Hi @cescoffier 

I did the rebase.

**Comment by cescoffier** [comment]
Looks good to me.

+1

File: src/test/java/io/vertx/core/impl/launcher/commands/WatcherTest.java [hunk 1]
```diff
@@ -175,6 +171,58 @@ public void testFileAdditionAndModificationInDirectory() throws IOException, Int
...
+    // create a file to be matched against "FOO.bar" pattern
+    File file = new File(root, "fOo.BAr");
+    file.createNewFile();
+    
+    if(ExecUtils.isWindows()) {      
+      // undeployment followed by redeployment. Windows is not case sensitive
+      waitUntil(() -> undeploy.get() == 1 && deploy.get() == 2);
+    }
+    else {
+      Thread.sleep(500);
```
Comment (vietj) [inline src/test/java/io/vertx/core/impl/launcher/commands/WatcherTest.java]: why do we have this pause here ?

**Comment by vietj** [comment]
I see lot of thread sleep in the tests. can we do without these ? in CI (cloudbees), such tests can fail because it runs slower than on our dev machines. so I'm not really a fan of having potential extra failures in the CI testsuite.

**Comment by cescoffier** [comment]
The watcher tests were / are passing on Cloudbees so, it should be fine. However, I agree we should find a different way to deal with these Thread.sleep. On way would be to use awaitility (https://github.com/jayway/awaitility) - a (test) library that relax the time pressure on the test and provide a very neat API. It's the lib I use in the other modules. However, it would require a CQ.

**Comment by msanguineti** [comment]
Hi, I've practically copied and pasted the code from Clement's tests in the same class. Just changed the conditions, obviously.

On 30 May 2016 09:43:41 EEST, Clement Escoffier <notifications@github.com> wrote:
>The watcher tests were / are passing on Cloudbees so, it should be
>fine. However, I agree we should find a different way to deal with
>these Thread.sleep. On way would be to use awaitility
>(https://github.com/jayway/awaitility) - a (test) library that relax
>the time pressure on the test and provide a very neat API. It's the lib
>I use in the other modules. However, it would require a CQ.
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/eclipse/vert.x/pull/1419#issuecomment-222422456

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Pull request merged**